torch/autograd/graph.py

```diff
@@ -439,27 +440,60 @@ def inner_hook(grad: torch.Tensor):
 
         return inner_hook
 
-    class Handle(RemovableHandle):
-        handles: Tuple[RemovableHandle, ...]
+    handles: List[RemovableHandle] = []
+    for i, t in enumerate(tensors):
+        handles.append(t.register_hook(get_inner_hook(i)))
 
-        def __init__(self, handles: Tuple[RemovableHandle, ...]):
-            self.handles = handles
+    return _MultiHandle(tuple(handles))
 
-        def remove(self):
-            for handle in self.handles:
-                handle.remove()
 
-        def __getstate__(self):
-            return self.handles
+def _register_any_hook(
+    tensors: Sequence[torch.Tensor],
```

Comment:
We have this hook only take in `Sequence[torch.Tensor]` and leave extracting the `torch.Tensor`s (that require gradient) from the forward outputs to FSDP.

Reply:
Hmm wdym, we also check if tensors that require grad below?

Reply:
Sorry for the confusion. I mainly meant that we do not allow arbitrary Python object as input to this function but rather just `Sequence[torch.Tensor]`:
https://github.com/pytorch/pytorch/blob/c19184cb61d12aea143586e776c5037bcb6b1022/torch/distributed/fsdp/_runtime_utils.py#L1377-L1383

Reply:
Ah gotcha, thanks for clarifying!